mysqldb/users.go

```diff
@@ -6,26 +6,37 @@ import (
 	"strings"
 
 	"github.com/artofimagination/mysql-user-db-go-interface/models"
+	"github.com/google/uuid"
 	"github.com/pkg/errors"
 
-	"github.com/google/uuid"
 	"golang.org/x/crypto/bcrypt"
 )
 
+// Defines the possible user query key names
+const (
+	ByEmail = "email"
+	ByID    = "id"
+)
+
 var ErrNoUserWithEmail = errors.New("There is no user associated with this email")
 
 var ErrSQLDuplicateUserNameEntryString = "Duplicate entry '%s' for key 'users.name'"
 var ErrSQLDuplicateEmailEntryString = "Duplicate entry '%s' for key 'users.email'"
 var ErrDuplicateUserNameEntry = errors.New("User with this name already exists")
+var ErrNoUserDeleted = errors.New("No user was deleted")
 
 var GetUserByEmailQuery = "select BIN_TO_UUID(id), name, email, password, BIN_TO_UUID(user_settings_id), BIN_TO_UUID(user_assets_id) from users where email = ?"
+var GetUserByIDQuery = "select BIN_TO_UUID(id), name, email, password, BIN_TO_UUID(user_settings_id), BIN_TO_UUID(user_assets_id) from users where id = UUID_TO_BIN(?)"
 
-// GetUserByEmail returns the user defined by the email.
-func (MYSQLFunctions) GetUserByEmail(email string, tx *sql.Tx) (*models.User, error) {
-	email = strings.ReplaceAll(email, " ", "")
+// GetUser returns the user defined by the key name and key value.
+// Key name can be either id or email.
+func (*MYSQLFunctions) GetUser(queryString string, keyValue interface{}, tx *sql.Tx) (*models.User, error) {
```

Comment:
I prefer the previous version where you used two different functions. Please, do not leave to the user the query parameters.
I also could send the following query with this kind of method directly to your database:
`DROP TABLE users;`

Reply:
Or if you really want only one method, then inside the method check the type of the key, and validate if it's an email address, if yes use email query, id query otherwise.

Reply:
Nah, clear sql injection mistake. I create a task for it to fix. I will revert back to separate methods.